### DISCUSSION TITLE
How to rectify "Failed to download xyz" (error 404) on some packages?

### DISCUSSION THREAD

**Discussion by buxbaumbixbybrae** [rickysarraf/apt-offline#200] [Q&A]
Hi All,

When attempting to get packages from the online system, a number of downloads fail (but most succeed); apt-offline prompts to run in verbose mode.
When run in verbose mode, the failures are error 404 and typically are packages that refer to ~20.04 in their name
The offline system is Ubuntu 20.04 with apt-offline 1.8.4. The online system is Linux mint 19.3 with apt-offline 1.8.4.
I believe that the offline system was successfully updated in April.

Commands:
```
offline$ sudo apt-offline set 20221024.sig --update --upgrade
online$ sudo apt-offline get 20221024.sig --bundle 20221025b.zip --verbose > logfile 2>&1
```
Error summary from non-verbose get:
```
ERROR: Failed to download zlib1g
ERROR: Failed to download libpython3.8
ERROR: Failed to download python3.8
ERROR: Failed to download libpython3.8-stdlib
ERROR: Failed to download python3.8-minimal
ERROR: Failed to download libpython3.8-minimal
ERROR: Failed to download rsync
ERROR: Failed to download libnetplan0
ERROR: Failed to download netplan.io
ERROR: Failed to download ubuntu-advantage-tools
ERROR: Failed to download openjdk-11-jre-headless
ERROR: Failed to download cloud-init
ERROR: Some items failed to download. Downloaded data may be incomplete
ERROR: Please run in verbose mode to see details about failed items
```
Attached signature file and verbose output.

What can I do to understand and rectify the issue please? Should I do --update separately to --upgrade (I don't think so), or upgrade to 1.8.5, or something else?

Thank you in advance
[20221024.sig.txt](https://github.com/rickysarraf/apt-offline/files/9859498/20221024.sig.txt)
[20221025-apt-offline-verbose.log](https://github.com/rickysarraf/apt-offline/files/9859500/20221025-apt-offline-verbose.log)

**Comment by buxbaumbixbybrae** [comment]
I have noticed that all the failed items are from http://archive.ubuntu.com/ubuntu/pool/main but that other packages from here succeed.

Trying to wget one of the failed links from the verbose output also returns 404 which is not surprising, I tried:
wget http://archive.ubuntu.com/ubuntu/pool/main/c/cloud-init/cloud-init_22.1-14-g2e17a0d6-0ubuntu1~20.04.3_all.deb

What I do not know is what is wrong with the link, nor how to update it so the download succeeds.
Searching for ubuntu apt error 404, one result suggested to substitute archive with old-releases, this does not work. another result suggested to rebuild /etc/apt/sources.list for the reverse reason ("20.04 is not old")

Why are some of the downloads from http://archive.ubuntu.com/ubuntu/pool/main failing?

I guess apt-offline is not the problem here, but I do not know how to get moving again.
Any clues, suggestions or assistance would be most welcome.

**Comment by rickysarraf** [comment]
Could you please split your first run command into 2 ?

>$ sudo apt-offline set 20221024.sig --update --upgrade

Please split it to 2 commands
>$ sudo apt-offline set 20221024.sig --update
and
>$ sudo apt-offline set 20221024.sig --upgrade

and then complete them separately.

What I mean is, 1st batch should be:

>$ sudo apt-offline set 20221024.sig --update
>$ sudo apt-offline get 20221024.sig --bundle update.zip
>$ sudo apt-offline install update.zip

2nd batch should be:

>$ sudo apt-offline set 20221024-upgrade.sig --upgrade
>$ sudo apt-offline get 20221024-upgrade.sig --bundle upgrade.zip
>$ sudo apt-offline install upgrade.zip


That should give you more consistent/deterministic output.


Note that the `404` error is purely on the remote Ubuntu server.
